shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobScheduler.java

```diff
@@ -69,6 +63,17 @@ public void stop() {
             each.stop();
             each.close();
         }
+        // TODO clean up should be done after the task is complete.
+        try {
+            TimeUnit.SECONDS.sleep(1);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        log.info("almost finished, preparer cleanup, job {}", jobContext.getJobId());
+        RuleAlteredJobPreparer jobPreparer = jobContext.getJobPreparer();
+        if (null != jobPreparer) {
+            jobPreparer.cleanup(jobContext);
+        }
```

Comment:
`cleanup` should be called only when job done and successful.

Reply:
Currently, the replication slot is deleted only after the task is complete